src/btree/bt_ret.c

```diff
@@ -49,6 +41,14 @@ __key_return(WT_CURSOR_BTREE *cbt)
             return (0);
         }
 
+        /* If the cursor references a WT_INSERT item, take its key. */
+        if (cbt->ins != NULL) {
+            WT_ASSERT(CUR2S(cbt), cbt->compare != 0);
+            cursor->key.data = WT_INSERT_KEY(cbt->ins);
+            cursor->key.size = WT_INSERT_KEY_SIZE(cbt->ins);
+            return (0);
+        }
```

Comment:
I'd move this back to where it was before, given that it doesn't cost us performance to do it there, and just in case there's another case we haven't caught.

I don't care much, though -- whatever you think is best!

Reply:
Agreed. Reverted to original code.